parser/worker.go

```diff
@@ -88,17 +84,9 @@ func (w Worker) process(height int64) error {
 	if height == 0 {
 		cfg := config.Cfg.Parser
 
-		var genesis *tmtypes.GenesisDoc
-		if strings.TrimSpace(cfg.GenesisFilePath) != "" {
-			genesis, err = w.getGenesisFromFilePath(cfg.GenesisFilePath)
-			if err != nil {
-				return err
-			}
-		} else {
-			genesis, err = w.getGenesisFromRPC()
-			if err != nil {
-				return err
-			}
+		genesis, _, err := GetGenesisDocAndState(cfg.GenesisFilePath, w.node)
```

Comment:
Instead of not making use of the returned `genesisState` map, can we instead pass it to `HandleGenesis` so that it's not necessary for it to re-decode it?